Zabbix

Whytey opened the pull request:
**Description:**
PR for introducing Zabbix Component and Zabbix Trigger Senors for interfacing with the Zabbix monitoring platform (http://www.zabbix.com/).  Intention is to add further Zabbix Sensors (for retrieving specific item values) and also allow publishing of Home Assistant state data to Zabbix monitoring instances.

**Related issue (if applicable):** 
N/A

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** 
home-assistant/home-assistant.github.io#1596

**Example entry for `configuration.yaml` (if applicable):**
```yaml
zabbix:
  host: 192.168.0.100
  username: Admin
  password: *****   

sensor:

  # Get total active trigger count for two hosts.
  - platform: zabbix         
    triggers:
      hostids: 
        - 10051
        - 10084

  # Get total active trigger count for all hosts.
  - platform: zabbix
    triggers:   

  # Get active trigger count for the two hosts, each as separate sensors.
  - platform: zabbix
    triggers:
      hostids: 
        - 10051
        - 10084
      individual: true

  # Get active trigger count for single host, and apply specified name to the sensor.
  - platform: zabbix 
    triggers:
      name: Test Name
      hostids: [10051]
      individual: true
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

Whytey committed:
initial working zabbix sensor

Whytey committed:
initial working zabbix sensor

Whytey committed:
Merge remote-tracking branch 'origin/master' into zabbix

Conflicts:
	homeassistant/const.py

merge in upstream

Whytey committed:
Merge remote-tracking branch 'upstream/master' into zabbix

Conflicts:
	homeassistant/const.py

merge upstream...I think

Whytey committed:
Working with multiple options for the sensor

Whytey committed:
Working with multiple options for the sensor

Whytey committed:
Tested code.  Ready to merge.

Whytey committed:
Tested code.  Ready to merge.

auduny committed:
Remove libtelldus-core-dev from Dockerfile (#4878)

Remove unnecessary  libtelldus-core-dev from Dockerfile . Ref https://github.com/home-assistant/home-assistant/pull/4680#issuecomment-266006310

pvizeli committed:
Migrate REST switch to async (#4517)

* Migrate REST switch to async

* Update rest.py

* Address comments from paulus

molobrakos committed:
Update __init__.py (#4877)

Cleaner exit by not throwing exception if server was not set during initialization of component (ref https://github.com/home-assistant/home-assistant/pull/4866)

Commit:
Added a volume set option and autodiscovery functions to Denon AVR rece… (#4845)

* Added Volume Set option and autodiscovery functions to Denon AVR receivers

* Corrected issues in SSDP discovery and in case no host could be discovered

* Corrected discovery handling / added denonavr to discovery platform

* No needless discoveries anymore / add_devices() with list instead of loop

balloob commented on homeassistant/const.py:
```diff
@@ -2,7 +2,7 @@
 MAJOR_VERSION = 0
 MINOR_VERSION = 34
-PATCH_VERSION = '2'
+PATCH_VERSION = '0.dev0'
```
Please do not include this change

balloob commented on homeassistant/const.py:
```diff
@@ -360,6 +360,7 @@
 HTTP_HEADER_CACHE_CONTROL = 'Cache-Control'
 HTTP_HEADER_EXPIRES = 'Expires'
 HTTP_HEADER_ORIGIN = 'Origin'
+HTTP_HEADER_X_FORWARDED_FOR = 'X-Forwarded-For'
```
Please do not include this change

Whytey replied:
I wish I could get rid of this change.  I reset any change in my branch by `git checkout upstream/master homeassistant/const.py` and it resolved the above conflict but wont remove this.  Not sure where the change came from :(

Whytey replied:
Hi @balloob , I did a blame and this line was added some time ago:
```
e73634e6 (hexa-                2016-11-13 01:14:39 +0100 367) HTTP_HEADER_X_FORWARDED_FOR = 'X-Forwarded-For'
```

All other checks and conflicts are resolved.  I think it is good to go.  Let me know if not.

Thanks,
Whytey

balloob commented on requirements_all.txt:
```diff
@@ -603,3 +603,6 @@ yahooweather==0.8
 # homeassistant.components.zeroconf
 zeroconf==0.17.6
+
+# homeassistant.components.zabbix
```
Please add requirements by following instructions in the PR

balloob commented on homeassistant/components/zabbix.py:
```diff
@@ -0,0 +1,58 @@
+    username = conf.get(CONF_USERNAME, None)
+    password = conf.get(CONF_PASSWORD, None)
+
+    global ZAPI
```
Please don't use global. Use the `hass.data` dictionary  instead

balloob commented on homeassistant/components/zabbix.py:
```diff
@@ -0,0 +1,58 @@
+
+    global ZAPI
+    ZAPI = ZabbixAPI(url)
+    ZAPI.login(username, password)
```
Can you verify that the login is correct here ? In that case please return `False`

balloob commented on homeassistant/components/sensor/zabbix_triggers.py:
```diff
@@ -0,0 +1,154 @@
+"""
```
Please rename this file to `zabbix` so it is the same as the component.

Whytey replied:
Hi @balloob.  Thanks for the review, will fix up all identified errors.  

For this comment, since there will be additional types of Zabbix sensor in the future (Zabbix Item Value) for example, wouldn't we need to separate these out into individual python files?  Also, doesn't the filename need to match the 'platform' type in the configuration file?

balloob commented on homeassistant/components/sensor/zabbix_triggers.py:
```diff
@@ -0,0 +1,154 @@
+        # Individual sensor per host
+        if not hostids:
+            # We need hostids
+            _LOGGER.critical("If using 'individual', must specify hostids")
```
Log level `error`. It's not critical for the execution of the HA core.

balloob commented on homeassistant/components/sensor/zabbix_triggers.py:
```diff
@@ -0,0 +1,154 @@
+            return False
+
+        for hostid in hostids:
+            _LOGGER.info("Creating Zabbix Sensor: " + str(hostid))
```
This seems very verbose to me. Especially since all state changes will also be logged

balloob commented on homeassistant/components/sensor/zabbix_triggers.py:
```diff
@@ -0,0 +1,154 @@
+        _LOGGER.debug("Updating ZabbixTriggerCountSensor: " + str(self._name))
+        triggers = self._call_zabbix_api()
+        self._state = len(triggers)
+        self._attributes['Last Update'] \
```
This won't be necessary. States already have a last_updated attribute

molobrakos committed:
bugfix: is_on is a property (#4889)

tchellomello committed:
Fixes issues #4844 to avoid traceback when self.rest.data is None (#4886)

6-12-09 18:12:30 homeassistant.core: Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/tasks.py", line 237, in _step
    result = next(coro)
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/helpers/entity_component.py", line 386, in _update_entity_states
    yield from update_coro
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/helpers/entity.py", line 240, in async_update_ha_state
    self._attr_setter('entity_picture', str, ATTR_ENTITY_PICTURE, attr)
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/helpers/entity.py", line 307, in _attr_setter
    value = getattr(self, name)
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/components/sensor/wunderground.py", line 176, in entity_picture
    url = self.rest.data['icon_url']
TypeError: 'NoneType' object is not subscriptable

Commit:
Update frontend

balloob committed:
Fix hue groups on older hubs (#4884)

Danielhiversen committed:
Bug in rpi_camera

valentinalexeev committed:
[media_player.sonos] Source selection from favorites (#4804)

pvizeli committed:
TTS add google language list for config check (#4912)

* Add config check for language

* update default

* move language from component to platform

* fix lint

Commit:
Update flux led lib

Danielhiversen committed:
Merge pull request #4907 from home-assistant/rpi_camera

Bug in rpi_camera

Danielhiversen committed:
Merge pull request #4924 from home-assistant/flux_lib

Update flux led lib

Whytey committed:
Fixed up most review comments

Whytey committed:
Fixed up most review comments

Whytey committed:
Last of the review comments addressed

Whytey committed:
Last of the review comments addressed

Whytey committed:
Last of the review comments addressed

Whytey committed:
Last of the review comments addressed

Whytey committed:
Last of the review comments addressed

Whytey committed:
Last of the review comments addressed

Whytey committed:
Really, the last of the fixups

Whytey committed:
Really, the last of the fixups

Whytey committed:
Really, the last of the fixups

Whytey committed:
Really, the last of the fixups

lwis committed:
[media_player.sonos] Bugfix, initalise source_name. (#4911)

pvizeli committed:
Homematic faster update with async (#4929)

pvizeli committed:
Protect add_job (#4932)

joopert committed:
kodi fanart fix basic auth (#4930)

Danielhiversen committed:
Bug fix for #4903 (#4927)

joerocklin committed:
Add nest hvac state (#4810)

* Add basic property details for Nest hvac_state

* Add the hvac_state sensor

* Update requirements and remove trailing whitespace

Clean up the multiline docstring

Adding a space between summary and description

* Removing the hvac_state as a property on the nest climate

* Update nest.py

jabesq committed:
Fix for GTFS sensor (#4828)

* Fix for GTFS sensor

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* GTFS fix

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

Danielhiversen committed:
Broadlink sensor and switch (#4834)

* Broadlink sensor and switch

* broadlink logging

* Use async

* style

* style

trisk committed:
Expose media volume as emulated Hue brightness (#4869)

* Allow virtual Hue bridge to set volume level of media_player entities
* Show correct states in all lights view

magas0 committed:
Panasonic viera fix (#4888)

* Removed return False so the Panasonic Viera TV can be added even if it doesn't connect

* Removed return False so the Panasonic Viera TV can be added even if it doesn't connect

* Removed return False so the Panasonic Viera TV can be added even if it doesn't connect

* Remove try/except to connect to the TV

* Update panasonic_viera.py

* Update panasonic_viera.py

fabaff committed:
Clean-up (#4894)

balloob committed:
Fix synologydsm (#4895)

fabaff committed:
Avoid TypeError for state (#4897)

magicus committed:
Add support for new netdisco detection of Samsung Smart TV. (#4925)

emlove committed:
Setup DarkSky platform when offline during init (#4919)

* Setup DarkSky platform when offline during init

* Fail setup_platform if fetch was unsuccessful

molobrakos committed:
initialize self._last_brightness (#4917)

roidayan committed:
Fix webostv component to accept any custom sources (#4915)

Updated the schema check to accept any string
Search custom sources in app title and app id
The makes the short list redundant and thus removed
Tested by adding livetv, netflix, youtube, makovod and others
This is also compatible with the list that was supported till now
so current users won't see any difference.

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

nkgilley committed:
add manual option to prevent scheduled tests. (#4906)

pvizeli committed:
Add voicerss for TTS (#4916)

* Add voicerss for TTS

* add unittests

* fix tests

* fix status bug in google/voicerss

* remove ssl

balloob committed:
Cast fix (#4939)

* Update frontend

* Fix exception on cast startup

Whytey committed:
Trying to pull in latest upstream changes

Whytey committed:
Fixup lint error

balloob commented:
Can you clean up your branch so that it only contains your commits? It looks like you have mixed a few commits in from somewhere.

Whytey commented:
Closing this PR in favour of #5297
